runtime/doc/api.txt

```diff
@@ -3228,28 +3228,20 @@ nvim_open_win({buffer}, {enter}, {config})                   *nvim_open_win()*
                       `eob` flag of 'fillchars' to a space char, and clearing
                       the |hl-EndOfBuffer| region in 'winhighlight'.
                   • border: Style of (optional) window border. This can either
-                    be a string or an array. The string values are
-                    • "none": No border (default).
-                    • "single": A single line box.
-                    • "double": A double line box.
-                    • "rounded": Like "single", but with rounded corners
-                      ("╭" etc.).
-                    • "solid": Adds padding by a single whitespace cell.
-                    • "shadow": A drop shadow effect by blending with the
-                      background.
-                    • If it is an array, it should have a length of eight or
-                      any divisor of eight. The array will specify the eight
-                      chars building up the border in a clockwise fashion
-                      starting with the top-left corner. As an example, the
-                      double box style could be specified as: >
-                      [ "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" ].
+                    be a string or an array. The string values are the same as
+                    those described in |winborder|. If it is an array, it
+                    should have a length of eight or any divisor of eight. The
+                    array will specify the eight chars building up the border
+                    in a clockwise fashion starting with the top-left corner.
+                    As an example, the double box style could be specified as: >
+                     [ "╔", "═" ,"╗", "║", "╝", "═", "╚", "║" ].
```

Comment:
'winborder' probably should also support this array form, but that can wait until @famiu's work

Reply:
> 'winborder' probably should also support this array form, but that can wait until @famiu's work. Or maybe it can be a comma-separated option?

We could do something similar to what `'fillchars'` does

Reply:
> comma-separated should be fine `set winborder=╔, ═ ,╗, ║, ╝, ═, ╚, ║ `. Similar to fillchars, it means that the user must declare the field name for each position like `set winborder=topleft:╔, top:═...` or i understand wrong 🤔

Yes that is what I suggested, since a list order may seem arbitrary. The `fillchars` syntax has the disadvantage of being unnecessarily verbose, but it follows an existing precedent so that's a plus imo.

Reply:
I think this should be implemented after we support `array/dict` option vlaues. when border is an array it has too many extension possibilities in `nvim_oepn_win`.. currently it is also possible to support all the formats described in `nvim_open_win`, but this may make the code a bit complicated and not very meaningful. If we have an explicit markup in the option value such as =[xx] it may be much easier.  or other smart way which i missed ?